app/updatemanager.go

```diff
@@ -386,6 +409,81 @@ func (c *ControlMapPool) ClearExpired() {
 		},
 	)
 	c.Pool = newPool
+
+	c.saveToStore()
+}
+
+type controlMapPoolDbFormat struct {
+	Active  []*UpdateControlMap `json:"active"`
+	Expired []*UpdateControlMap `json:"expired"`
+}
+
+// Must only be called from functions that have already locked the mutex.
+func (c *ControlMapPool) saveToStore() {
+	if c.store == nil {
+		// Saving not enabled. This should not happen outside tests.
+		return
+	}
+
+	active, expired := c.getAll()
+
+	toSave := controlMapPoolDbFormat{
+		// Intentionally not using label assignment here. We want to
+		// know if we're missing any fields.
```

Comment:
Then I'm afraid I've misunderstood, you want it to warn you when you're missing a field in the struct?

Reply:
Exactly. Since this is saved in the database, it might potentially be long lived and bugs in this area might "survive" from one mender version to the next even if fixed in the new version. That's why I want to be really pedantic in checking everything.